tesseract_environment/test/tesseract_environment_unit.cpp

```diff
@@ -151,6 +221,29 @@ Environment::Ptr getEnvironment(bool register_default_contact_managers = false)
   return env;
 }
 
+template <typename S>
+void runEnvInitFailuresTest()
+{
+  auto env = std::make_shared<Environment>();
+  EXPECT_TRUE(env != nullptr);
+  EXPECT_EQ(0, env->getRevision());
+  EXPECT_FALSE(env->reset());
```

Comment:
I didn't realize an uninitialized env returned false if you called reset. That makes sense, but we might want to update the doxygen.

Reply:
I was not sure on this one. I guess it should probably return true?

Reply:
The doxygen says it resets to the initialized state, so returning false if it isn't initialized makes sense to me. I was just saying we should specify in documentation.

Reply:
Documented.